packages/third-parties/src/google/gtm.tsx

```diff
@@ -0,0 +1,64 @@
+'use client'
+// TODO: Evaluate import 'client only'
+import React from 'react'
+import { useCallback } from 'react'
+import Script from 'next/script'
+
+declare global {
+  interface Window {
+    dataLayer?: Object[]
+    [key: string]: any
+  }
+}
+
+type GTMParams = {
+  gtmId: string
+  dataLayer: string[]
+  dataLayerName: string
+  auth: string
+  preview: string
+}
+
+let currDataLayerName = 'dataLayer'
+
+export function GoogleTagManager(props: GTMParams) {
+  const { gtmId, dataLayerName = 'dataLayer', auth, preview, dataLayer } = props
+
+  currDataLayerName = dataLayerName
+
+  const gtmLayer = dataLayerName !== 'dataLayer' ? `$l=${dataLayerName}` : ''
+  const gtmAuth = auth ? `&gtm_auth=${auth}` : ''
+  const gtmPreview = preview ? `&gtm_preview=${preview}&gtm_cookies_win=x` : ''
+
+  return (
+    <>
+      <Script
+        id="_next-gtm-init"
+        dangerouslySetInnerHTML={{
+          __html: `
+      (function(w,l){
+        w[l]=w[l]||[];
+        w[l].push({'gtm.start': new Date().getTime(),event:'gtm.js'});
+        ${dataLayer ? `w[l].push(${JSON.stringify(dataLayer)})` : ''}
+      })(window,'${dataLayerName}');`,
+        }}
+      ></Script>
+      <Script
+        id="_next-gtm"
+        src={`https://www.googletagmanager.com/gtm.js?id=${gtmId}${gtmLayer}${gtmAuth}${gtmPreview}`}
+      ></Script>
+    </>
+  )
+}
+
+export function useGoogleTagManager() {
+  const sendData = useCallback((data: Object) => {
+    if (window[currDataLayerName]) {
+      window[currDataLayerName].push(data)
+    } else {
+      console.warn(`dataLayer ${currDataLayerName} does not exist`)
+    }
+  }, [])
+
+  return { sendData }
+}
```

Comment:
IMHO I don't see why this should be a React Hooks. The `sendData` function doesn't have any dependency (the dependency array has a length of 0), which means it can live outside of React / Next.js without `useCallback`.

Why not just:

```ts
export const gtag = (data: Object) => {
  if (window[currDataLayerName]) {
    window[currDataLayerName].push(data)
  } else {
    console.warn(`dataLayer ${currDataLayerName} does not exist`)
  }
}
```

Reply:
That's a good point. I've updated the PR. Thanks